### ISSUE TITLE
Beautifier makes JavaScript syntax invalid

### ISSUE THREAD

**Issue by supaiku-o** [beautify-web/js-beautify#876]
Original: https://github.com/Glavin001/atom-beautify/issues/810

The beautifier makes this code invalid:

> .pragma library
> .import "../Constants.js" as Constants
>
>var MyClass = {
>
>    something: function() {
>        console.log(Constants.Bla);
>   }
>}

The above code is pretty common among QML.
http://doc.qt.io/qt-5/qtqml-javascript-imports.html#importing-a-javascript-resource-from-another-javascript-resource

**Comment by bitwiseman** [comment]
But `.import` isn't javascript syntax, is it?

**Comment by supaiku-o** [comment]
Yes, that's correct. But I think it should be supported, because there is no alternative besides using QtCreator.

If that's not good enough, how about exposing it via a "QML compatibilty" setting. This way nothing is changed by default and everyone can be happy. :p

**Comment by bitwiseman** [comment]
No small undertaking from what I can see. I'll keep it as an enhancement.
